### PR TITLE
Add support to install v2 and any other cosign release candidate

### PR DESCRIPTION
**Pull request by hectorj2f**
Signed-off-by: Hector Fernandez <hector@chainguard.dev>

#### Summary

Enable installing release candidates. I am starting to use cosign v2 release candidate and I could not test it using the cosign-installer due to the regular expressions and other settings.
Please, let me know if we want to enable the installation of release candidates. Otherwise I am happy to close this PR.

### CODE REVIEW

**Comment by hectorj2f** [comment]
I am investigating why the cosign v2.0.0-rc throws a NotFound error when checking the tlog while older versions don't. You can see this issue when verifying the v0.6.0 artifact.

File: action.yml [hunk 1]
```diff
@@ -10,7 +10,7 @@ inputs:
   cosign-release:
     description: 'cosign release version to be installed'
     required: false
-    default: 'v1.13.1'
+    default: 'v2.0.0-rc.0'
```
Comment (k4leung4) [inline action.yml:13]: does it make sense to default to a RC build?

not sure if users will expect the default to be a release candidate. i think it we should support letting users specify RC builds if they choose do so, but not default them into it.
and similarly for the bootstrap version, we should choose a non-RC version for it imho.

Comment (hectorj2f) [reply action.yml:13]: I just wanted to change the code to point to latest available version.

Comment (hectorj2f) [reply action.yml]: I'll revert it, but I feel once we get our v2 stable version ready, we should change it.

Comment (k4leung4) [reply action.yml]: agree, once v2 stable version, we should change it to that.

thanks

**Review by k4leung4 — DISMISSED** [review]

**Comment by hectorj2f** [comment]
This PR is ready for a review. ping @cpanato

**Review by cpanato — APPROVED** [review]

**Comment by cpanato** [comment]
thank you

**Comment by hectorj2f** [comment]
Thanks @cpanato and @k4leung4 ❤️

**Pull request merged**